### PR TITLE
MM-13622 MM-13620 Adding client4 methods, reducers, actions, and selectors for bot accounts.

### PR DESCRIPTION
**Pull request by crspeller**
#### Summary
Added for bot accounts:
- Client4 methods
- Reducers
- Actions
- Selectors

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-13620
https://mattermost.atlassian.net/browse/MM-13622

#### Checklist
- [x] Ran `make check-style` to check for style errors (required for all pull requests)
- [x] Ran `make test` to ensure unit tests passed
- [x] Added or updated unit tests (required for all new features)
- [x] All new/modified APIs include changes to the drivers

#### Test Information
This PR was tested on: ArchLinux

### CODE REVIEW

File: src/actions/bots.test.js [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+
+    beforeEach(async () => {
+        store = await configureStore();
+    });
+
+    afterAll(async () => {
+        await TestHelper.tearDown();
+    });
+
+    it('loadBots', async () => {
```
Comment (levb) [inline src/actions/bots.test.js]: nit: should this test load multiple (2?) fake bots?

File: src/actions/bots.test.js [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+    });
+
+    it('disableBot', async () => {
+        const bot = TestHelper.fakeBot();
+        nock(Client4.getBotRoute()).
+            post('').
+            reply(200, bot);
+        await BotActions.createBot(bot)(store.dispatch, store.getState);
+
+        bot.delete_at = 1507840900065;
```
Comment (levb) [inline src/actions/bots.test.js]: nit: maybe a quick comment explaining the relationship between delete_at and `disableBot`?

File: src/actions/bots.test.js (hunk 1 above)
Comment (crspeller) [reply src/actions/bots.test.js]: fixed.

File: src/actions/bots.test.js (hunk 2 above)
Comment (crspeller) [reply src/actions/bots.test.js]: added.

File: src/actions/bots.test.js [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+    });
+
+    it('loadBots', async () => {
+        const bots = [TestHelper.fakeBot(), TestHelper.fakeBot()];
+        nock(Client4.getBotsRoute()).
+            get('').
+            query(true).
+            reply(201, bots);
+
+        await BotActions.loadBots()(store.dispatch, store.getState);
```
Comment (hmhealey) [inline src/actions/bots.test.js]: We should actually be calling `dispatch` on the result of these action creators.

```javascript
await store.dispatch(BotActions.loadBots());
```

Both ways of calling this work the same with async actions, but the way you're doing it doesn't work with plain redux actions. Also, doing it this way saves having to pass `getState` explicitly

File: src/reducers/entities/bots.js [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+        }
+        return nextState;
+    }
+    case BotTypes.RECEIVED_BOT_ACCOUNT: {
+        const bot = action.data;
+        const nextState = {...state};
+        nextState[bot.user_id] = bot;
+        return nextState;
+    }
+    default:
```
Comment (hmhealey) [inline src/reducers/entities/bots.js]: We should probably have a case here to clear the state on logout

File: src/actions/bots.test.js (hunk 3 above)
Comment (crspeller) [reply src/actions/bots.test.js]: Fixed.

File: src/reducers/entities/bots.js (hunk 4 above)
Comment (crspeller) [reply src/reducers/entities/bots.js]: Added.

**Comment by crspeller** [comment]
@hmhealey Ready for re-review

**Pull request merged**